src/librustc_codegen_ssa/back/symbol_export.rs

```diff
@@ -196,7 +196,7 @@ fn exported_symbols_provider_local<'a, 'tcx>(tcx: TyCtxt<'a, 'tcx, 'tcx>,
                                  })
                                  .collect();
 
-    if tcx.sess.entry_fn.borrow().is_some() {
+    if tcx.entry_fn(LOCAL_CRATE).is_some() {
         let exported_symbol = ExportedSymbol::NoDefId(SymbolName::new("main"));
```

Comment:
Hardcoding the entry point to `main` seems suspicious, not a fault of this PR though. We should probably open an issue for that.

Reply:
Filed https://github.com/rust-lang/rust/issues/57591.